sql/src/main/java/org/apache/druid/sql/calcite/planner/CalciteRulesManager.java

```diff
@@ -262,6 +262,7 @@ private Program buildBaseRuleSetProgram(PlannerContext plannerContext)
     builder.addMatchLimit(CalciteRulesManager.HEP_DEFAULT_MATCH_LIMIT);
     builder.addGroupBegin();
     builder.addRuleCollection(baseRuleSet(plannerContext));
+    builder.addRuleInstance(CoreRules.UNION_MERGE);
```

Comment:
I feel the name of the method is no longer accurate (it's going beyond `baseRuleSet`). It appears this is only used in decoupled mode-- is that correct? If so some comments about that, or reflecting it in the name, would be useful.

Haven't read the rest of the patch as yet, so this is just a single comment rather than a review.

Reply:
I wanted to separate basic optimiazation rules from the conversion rules - it will be usefull to be able to enable/tweak rules without altering the other planning rules.

It kinda already made one things easier: I was able to just throw in `UNION_MERGE` - without having to worry about disturbing existing plan differences.

there will be some more deeper tweaking will be needed as `AGGREGATE_REMOVE` and `AGGREGATE_CASE_TO_FILTER` doesn't play nicely together - which induces some plan differences

I can rename it to `buildDecoupledLogicalOptimizationProgram` - does that sound better?